gno.land/pkg/sdk/vm/builtins_test.go

```diff
@@ -0,0 +1,61 @@
+package vm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParamsRestricted(t *testing.T) {
+	env := setupTestEnv()
+	params := NewSDKParams(env.vmk, env.ctx)
+	params.SetCurRealmPath("gno.land/r/foo")
+
+	testCases := []struct {
+		name        string
+		setFunc     func()
+		expectedMsg string
+	}{
+		{
+			name: "SetString should panic",
+			setFunc: func() {
+				params.SetString("name.string", "foo")
```

Comment:
these tests are misleading. 
the expected format is <realm>.<paramname>.<type>. 
so if anything, what should be passed in is something like "gno.land/r/foo.name.string". 
which also means params.SetCurRealmPath("gno.land/r/foo") is an unnecessary call.

See my other comments about curRealmPath.

Reply:
agree. fixed